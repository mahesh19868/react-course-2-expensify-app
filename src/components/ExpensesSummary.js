import React from 'react';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import getExpensesTotal from '../selectors/expenses-total';
import numeral from 'numeral';


export const ExpensesSummary = ({expenseCount,expenseTotal}) => {
    const messageWord = (expenseCount === 1) ? 'expense': 'expenses';
    return (
        <div>
           <h1>Viewing {expenseCount} {messageWord} totalling {expenseTotal} </h1>
        </div>
    );   
}

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters);
    const expenseTotal = getExpensesTotal(visibleExpenses);
    return {
        expenseTotal: numeral(expenseTotal / 100).format('$0,0.00'),         
        expenseCount: visibleExpenses.length            
    };
};

export default connect(mapStateToProps)(ExpensesSummary);