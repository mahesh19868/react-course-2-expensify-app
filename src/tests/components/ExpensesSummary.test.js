import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should render Expenses Summary with 1 expenses',() => {
    const wrapper = shallow(<ExpensesSummary expenseTotal={235} expenseCount={1} />);
    expect(wrapper).toMatchSnapshot();
})


test('should render Expenses Summary with multiple expenses',() => {
    const wrapper = shallow(<ExpensesSummary expenseTotal={2350} expenseCount={23} />);
    expect(wrapper).toMatchSnapshot();
})