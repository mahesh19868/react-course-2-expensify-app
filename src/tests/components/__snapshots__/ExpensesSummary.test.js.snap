// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Expenses Summary with 1 expenses 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    235
     
  </h1>
</div>
`;

exports[`should render Expenses Summary with multiple expenses 1`] = `
<div>
  <h1>
    Viewing 
    23
     
    expenses
     totalling 
    2350
     
  </h1>
</div>
`;
